src/components/pages/tx/transaction-details.tsx

```diff
@@ -6,39 +7,76 @@ import EthereumIcon from "@/components/lib/ethereum-icon";
 import { Card, CardContent } from "@/components/ui/card";
 import { Transaction } from "@/lib/types";
 import { formatGwei, formatEther } from "viem";
+import { useEffect, useState } from "react";
+import { l2PublicClient } from "@/lib/chains";
 
-const TransactionDetails = ({ transaction }: { transaction: Transaction }) => (
-  <Card>
-    <CardContent className="p-4">
-      <dl>
-        <DescriptionListItem title="Transaction Hash">
-          {transaction.hash}
-        </DescriptionListItem>
-        <DescriptionListItem title="Block">
-          <Link
-            className="text-primary hover:brightness-150"
-            href={`/block/${transaction.blockNumber}`}
-          >
-            {transaction.blockNumber.toString()}
-          </Link>
-        </DescriptionListItem>
-        <DescriptionListItem title="Timestamp">
-          <Clock className="mr-1 size-4" />
-          {formatTimestamp(transaction.timestamp)}
-        </DescriptionListItem>
-        <DescriptionListItem border title="Value">
-          <EthereumIcon className="mr-1 size-4" />
-          {formatEther(transaction.value)} ETH
-        </DescriptionListItem>
-        <DescriptionListItem title="Gas Price">
-          {formatGwei(transaction.gasPrice ?? BigInt(0))} Gwei
-          <span className="ml-1 text-muted-foreground">
-            ({formatEther(transaction.gasPrice ?? BigInt(0))} ETH)
-          </span>
-        </DescriptionListItem>
-      </dl>
-    </CardContent>
-  </Card>
-);
+const TransactionDetails = ({ transaction }: { transaction: Transaction }) => {
+  const [isContract, setIsContract] = useState<boolean | null>(null);
+
+  useEffect(() => {
```

Comment:
While the logic is good, we don't want it to happen client-side because we want to mirror how the explorer will work when the accompanying indexer is ready.
Move your logic server-side by adding an optional `isContractInteraction` property in the `Transaction` type declared in `@/lib/types`.
Populate this computed property using your client-side logic in the top-level Page component and pass the updated transaction object to this `TransactionDetails` component.
Do not display "Contract interaction" or "EOA transfer" next to the `to` address, instead modify the label to display "Interacted With (To):" in case of contract interaction.